### PR TITLE
Added user location button to Places Plugin's PlacePickerActivity

### LINKED ISSUE #973: Add device location button in Place Picker activity
Creating this ticket to track @androideveloper 's request for [a way to enable "My location" button in Place Picker page that zooms to user current location](https://github.com/mapbox/mapbox-plugins-android/pull/965#issuecomment-498043305)

Floating action button would be added to the [PlacePickerActivity](https://github.com/mapbox/mapbox-plugins-android/blob/master/plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java) XML layout and we'd use the Maps SDK's `LocationComponent` in the activity.

Screen Shot 2019-06-03 at 10 37 46 AM

### PR DESCRIPTION
**Pull request by langsmith**
Resolves #973 by adding a device location Floating Action Button to the `PlacePickerActivity`. The button is hooked into the Maps SDK's `LocationComponent`, so this pr also adds the `LocationComponent` setup to the `PlacePickerActivity`.

This pr also adds a `includeDeviceLocationButton()` option to `PlacePickerOptions.java`. If the `includeDeviceLocationButton()` method isn't used, the default is to _not_ include the button in the `PlacePickerActivity` UI.

cc @androideveloper, who requested this.

ezgif com-resize (2)

### CODE REVIEW

File: plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java [hunk 1]
```diff
@@ -123,25 +141,89 @@ public void onMapReady(final MapboxMap mapboxMap) {
...
+        public void onStyleLoaded(@NonNull Style style) {
+          if (options != null && options.includeDeviceLocationButton()) {
+            enableLocationComponent(style);
+          }
+        }
+      });
+    } else {
+      Toast.makeText(this, R.string.mapbox_plugins_place_picker_user_location_permission_not_granted,
+        Toast.LENGTH_LONG).show();
+      finish();
```
Comment (LukasPaczos) [inline plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java]: I don't think we should finish the picker activity when permissions are not granted, we should rather just hide the button.

File: plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java [hunk 2]
```diff
@@ -123,25 +141,89 @@ public void onMapReady(final MapboxMap mapboxMap) {
...
+      mapboxMap.getStyle(new Style.OnStyleLoaded() {
+        @Override
+        public void onStyleLoaded(@NonNull Style style) {
+          if (options != null && options.includeDeviceLocationButton()) {
+            enableLocationComponent(style);
+          }
+        }
+      });
+    } else {
+      Toast.makeText(this, R.string.mapbox_plugins_place_picker_user_location_permission_not_granted,
```
Comment (LukasPaczos) [inline plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java]: This toast doesn't seem necessary.

File: plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java [hunk 3]
```diff
@@ -207,6 +289,31 @@ public void onClick(View view) {
...
+  /**
+   * Bind the device location Floating Action Button to this activity's UI and move the
+   * map camera if the button's clicked.
+   */
+  private void addUserLocationButton() {
+    userLocationButton.show();
+    userLocationButton.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        if (mapboxMap.getLocationComponent().getLastKnownLocation() != null) {
```
Comment (LukasPaczos) [inline plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java]: You can assign last know location to a local field and re-use that across this method.

File: plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java [hunk 4]
```diff
@@ -207,6 +289,31 @@ public void onClick(View view) {
...
+          mapboxMap.moveCamera(CameraUpdateFactory.newCameraPosition(
```
Comment (LukasPaczos) [inline plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java]: How about `animateCamera` with some short duration instead?

File: plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java (hunk 2 above)
Comment (langsmith) [reply plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java]: Removed

File: plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java (hunk 1 above)
Comment (langsmith) [reply plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java]: Fixed

File: plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java (hunk 3 above)
Comment (langsmith) [reply plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java]: Fixed

File: plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java (hunk 4 above)
Comment (langsmith) [reply plugin-places/src/main/java/com/mapbox/mapboxsdk/plugins/places/picker/ui/PlacePickerActivity.java]: Fixed

```java
if (mapboxMap.getLocationComponent().getLastKnownLocation() != null) {
          Location lastKnownLocation = mapboxMap.getLocationComponent().getLastKnownLocation();
          mapboxMap.animateCamera(CameraUpdateFactory.newCameraPosition(
            new CameraPosition.Builder()
              .target(new LatLng(lastKnownLocation.getLatitude(),
                  lastKnownLocation.getLongitude()))
              .zoom(17.5)
              .build()
          ),1400);
        }
```

**Comment by androideveloper** [comment]
Thanks!

**Pull request merged**